src/client/multiplayer/multiplayer.js

```diff
@@ -139,6 +139,10 @@ export class Multiplayer {
   updateGameID(id) {
     this.gameID = this.gameName + ':' + id;
 
+    const action = ActionCreators.reset();
```

Comment:
This was added because changing a player or game should reset the internal state and re-sync with the server

Reply:
Shouldn't the `this.socket.emit` below take care of this?

Reply:
@nicolodavis in this instance, when the sync got sent back from the server to the client, the client rejects the sync because the incoming stateID is often seen as earlier/lower than the existing state's stateID, so it's discarded. That appears to be a bug, because we shouldn't compare to an existing stateID if we have switched games, as the new game's state might be totally different, so this comparison no longer makes sense. 

The first thing that came to mind is that we should flush state when we change players or games because in either case the existing state we have shouldn't be used anymore. We could do something different, like change the sync comparison, but I wasn't sure if that was the right move here. What do you think?

Reply:
Yeah, we should change the sync comparison instead. I can make that change.